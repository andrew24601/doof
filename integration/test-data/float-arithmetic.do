// Test float arithmetic operations
// Tests: ADD_FLOAT, SUB_FLOAT, MUL_FLOAT, DIV_FLOAT, LOADK

function main(): int {
    let a: float = 10.5;
    let b: float = 2.5;
    
    let sum: float = a + b;        // ADD_FLOAT
    let diff: float = a - b;       // SUB_FLOAT
    let product: float = a * b;    // MUL_FLOAT
    let quotient: float = a / b;   // DIV_FLOAT
    
    println(sum);      // Should print 13
    println(diff);     // Should print 8
    println(product);  // Should print 26.25
    println(quotient); // Should print 4.2
    
    return 0;
}
