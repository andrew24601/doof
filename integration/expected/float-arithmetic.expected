13
8
26.25
4.2
